File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/sound/SamplesTarget.java
/**
 * @file
 * @brief Samples target interface
 * @author vitamin.caig@gmail.com
 */

package app.zxtune.sound;

public interface SamplesTarget {

  /**
   * @return target sample rate in Hz
   */
  int getSampleRate();

  /**
   * @return buffer size in samples
   */
  int getPreferableBufferSize();

  /**
   * Initialize target
   */
  void start() throws Exception;

  /**
   * @param buffer sound data in S16/stereo/interleaved format
   */
  void writeSamples(short[] buffer) throws Exception;

  /**
   * Deinitialize target
   */
  void stop() throws Exception;

  /**
   * Release all internal resources
   */
  void release();
}

File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/sound/SamplesTarget.kt
/**
 * @file
 * @brief Samples target interface
 * @author vitamin.caig@gmail.com
 */
package app.zxtune.sound

import app.zxtune.Releaseable

interface SamplesTarget : Releaseable {
    /**
     * @return target sample rate in Hz
     */
    val sampleRate: Int

    /**
     * @return buffer size in samples
     */
    val preferableBufferSize: Int

    /**
     * Initialize target
     */
    @Throws(Exception::class)
    fun start()

    /**
     * @param buffer sound data in S16/stereo/interleaved format
     */
    @Throws(Exception::class)
    fun writeSamples(buffer: ShortArray)

    /**
     * Deinitialize target
     */
    @Throws(Exception::class)
    fun stop()
}
